lib/trento/domain/sap_system/commands/deregister_application_instance.ex

```diff
@@ -0,0 +1,17 @@
+defmodule Trento.Domain.Commands.DeregisterApplicationInstance do
+  @moduledoc """
+  Deregister (decommission) an application instance from the monitoring system.
+  """
+
+  @required_fields [
+    :instance_number,
+    :sap_system_id
+  ]
+
+  use Trento.Command
+
+  defcommand do
+    field :instance_number, :string
+    field :sap_system_id, Ecto.UUID
```

Comment:
We need to add the `deregistered_at` field in the commands as well, as we did with the host/cluster aggregate commands